ext/opcache/jit/zend_jit_x86.dasc

```diff
@@ -0,0 +1,9938 @@
+| 	.endif
+|.endmacro
+
+|.macro LOAD_BASE_ADDR, reg, base, offset
+||	if (offset) {
+|		lea reg, qword [Ra(base)+offset]
+||	} else {
+|		mov reg, Ra(base)
+||	}
+|.endmacro
+
+|.macro PUSH_BASE_ADDR, base, offset, tmp_reg
+||	if (offset) {
+|		lea tmp_reg, qword [Ra(base)+offset]
+|		push tmp_reg
+||	} else {
+|		push Ra(base)
+||	}
+|.endmacro
+
+|.macro EXT_CALL, func, tmp_reg
+|	.if X64
+||		if (IS_32BIT(dasm_end) && IS_32BIT(func)) {
+|			call qword &func
+||		} else {
+|			LOAD_ADDR tmp_reg, func
+|			call tmp_reg
+||		}
+|	.else
+|		call dword &func
+|	.endif
+|.endmacro
+
+|.macro EXT_JMP, func, tmp_reg
+|	.if X64
+||		if (IS_32BIT(dasm_end) && IS_32BIT(func)) {
+|			jmp qword &func
+||		} else {
+|			LOAD_ADDR tmp_reg, func
+|			jmp tmp_reg
+||		}
+|	.else
+|		jmp dword &func
+|	.endif
+|.endmacro
+
+|.macro LOAD_ZVAL_ADDR, reg, addr
+||	if (Z_MODE(addr) == IS_CONST_ZVAL) {
+|		LOAD_ADDR reg, Z_ZV(addr)
+||	} else if (Z_MODE(addr) == IS_MEM_ZVAL) {
+|   	LOAD_BASE_ADDR reg, Z_REG(addr), Z_OFFSET(addr)
+||	} else {
+||		ZEND_ASSERT(0);
+||	}
+|.endmacro
+
+|.macro PUSH_ZVAL_ADDR, addr, tmp_reg
+||	if (Z_MODE(addr) == IS_CONST_ZVAL) {
+|		PUSH_ADDR Z_ZV(addr), tmp_reg
+||	} else if (Z_MODE(addr) == IS_MEM_ZVAL) {
+|   	PUSH_BASE_ADDR Z_REG(addr), Z_OFFSET(addr), tmp_reg
+||	} else {
+||		ZEND_ASSERT(0);
+||	}
+|.endmacro
+
+|.macro GET_Z_TYPE_INFO, reg, zv
+|	mov reg, dword [zv+offsetof(zval,u1.type_info)]
+|.endmacro
+
+|.macro SET_Z_TYPE_INFO, zv, type
+|	mov dword [zv+offsetof(zval,u1.type_info)], type
+|.endmacro
+
+|.macro GET_ZVAL_TYPE_INFO, reg, addr
+||	ZEND_ASSERT(Z_MODE(addr) == IS_MEM_ZVAL);
+|	mov reg, dword [Ra(Z_REG(addr))+Z_OFFSET(addr)+offsetof(zval,u1.type_info)]
+|.endmacro
+
+|.macro SET_ZVAL_TYPE_INFO, addr, type
+||	ZEND_ASSERT(Z_MODE(addr) == IS_MEM_ZVAL);
+|	mov dword [Ra(Z_REG(addr))+Z_OFFSET(addr)+offsetof(zval,u1.type_info)], type
+|.endmacro
+
+|.macro GET_Z_PTR, reg, zv
+|	mov reg, aword [zv]
+|.endmacro
+
+|.macro SET_Z_PTR, zv, val
+|	mov aword [zv], val
+|.endmacro
+
+|.macro GET_ZVAL_PTR, reg, addr
+||	ZEND_ASSERT(Z_MODE(addr) == IS_MEM_ZVAL);
+|	mov reg, aword [Ra(Z_REG(addr))+Z_OFFSET(addr)]
+|.endmacro
+
+|.macro SET_ZVAL_PTR, addr, val
+||	ZEND_ASSERT(Z_MODE(addr) == IS_MEM_ZVAL);
+|	mov aword [Ra(Z_REG(addr))+Z_OFFSET(addr)], val
+|.endmacro
+
+|.macro GET_ZVAL_W2, reg, addr
+||	ZEND_ASSERT(Z_MODE(addr) == IS_MEM_ZVAL);
+|	mov reg, dword [Ra(Z_REG(addr))+Z_OFFSET(addr)+4]
+|.endmacro
+
+|.macro SET_ZVAL_W2, addr, val
+||	ZEND_ASSERT(Z_MODE(addr) == IS_MEM_ZVAL);
+|	mov dword [Ra(Z_REG(addr))+Z_OFFSET(addr)+4], val
+|.endmacro
+
+|.macro FPU_OP, fp_ins, addr
+||	if (Z_MODE(addr) == IS_CONST_ZVAL) {
+|		MEM_OP1 fp_ins, qword, Z_ZV(addr), r0
+||	} else if (Z_MODE(addr) == IS_MEM_ZVAL) {
+|   	fp_ins qword [Ra(Z_REG(addr))+Z_OFFSET(addr)]
+||	} else {
+||		ZEND_ASSERT(0);
+||	}
+|.endmacro
+
+|.macro FPU_GET_ZVAL_DVAL, addr
+|	FPU_OP fld, addr
+|.endmacro
+
+|.macro FPU_MATH, opcode, addr
```

Comment:
I think we should not support JIT if the CPU doesn't have at least SSE(2). This seems like an unnecessary complication for systems that you're only going to find in an antiquary...

Reply:
In general, I agree. Currently FPU code is not used anyway.